### DISCUSSION TITLE
this sqlexpress is error but i don't know  error where

### DISCUSSION THREAD

**Discussion by q1165600785** [sqlalchemy/sqlalchemy#9580] [Usage Questions]
def query_data_model_name(self, conn,model_name):
        stmt = select(antmod_one_one.c.id, antmod_one_one.c.model_name,
                       antmod_one_one.c.antmod_name,antmod_one_one.c.rad,
                       antmod_one_one.c.dh, antmod_one_one.c.L1_E, antmod_one_one.c.L1_N,
                       antmod_one_one.c.L1_U,antmod_one_one.c.L2_E,
                       antmod_one_one.c.L2_N, antmod_one_one.c.L2_U,
                       antmod_one_one.c.remark,
                      (func.group_concat(antmod_picture.c.picture_dir),
                       func.group_concat(model_picture.c.picture_dir)),
                      ).join_from(antmod_one_one, model_picture, antmod_one_one.c.model_name == model_picture.c.model,isouter=True).join_from(
            antmod_one_one, antmod_picture, antmod_one_one.c.antmod_name == antmod_picture.c.antmod,isouter=True).where(
            antmod_one_one.c.model_name.like("%" + model_name + "%")).group_by(antmod_one_one.c.id)
        qry = conn.execute(stmt, execution_options={"prebuffer_rows": True})
        return qry

Traceback (most recent call last):
  File "E:\python\anconda\envs\query_antmod_env\lib\site-packages\sqlalchemy\sql\base.py", line 1565, in __getattr__
    return self._index[key][1]
KeyError: 'model'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "I:/gamit/query_antmod_v3/data_base_z/database_connection/database_connection.py", line 189, in <module>
    result=get_opr.query_data_model_name(conn,"zz")
  File "I:/gamit/query_antmod_v3/data_base_z/database_connection/database_connection.py", line 57, in query_data_model_name
    ).join_from(antmod_one_one, model_picture, antmod_one_one.c.model_name == model_picture.c.model,isouter=True).join_from(
  File "E:\python\anconda\envs\query_antmod_env\lib\site-packages\sqlalchemy\sql\base.py", line 1567, in __getattr__
    raise AttributeError(key) from err
AttributeError: model

**Comment by q1165600785** [comment]
def query_data_model_name(self, conn,model_name):
        stmt = select(antmod_one_one.c.id, antmod_one_one.c.model_name,
                       antmod_one_one.c.antmod_name,antmod_one_one.c.rad,
                       antmod_one_one.c.dh, antmod_one_one.c.L1_E, antmod_one_one.c.L1_N,
                       antmod_one_one.c.L1_U,antmod_one_one.c.L2_E,
                       antmod_one_one.c.L2_N, antmod_one_one.c.L2_U,
                       antmod_one_one.c.remark,
                      func.concat_ws(func.group_concat(antmod_picture.c.picture_dir),
                    func.group_concat(model_picture.c.picture_dir)).label("picture_dir")
                      ).join_from(antmod_one_one, model_picture, antmod_one_one.c.model_name == model_picture.c.model,isouter=True
                     ).join_from(antmod_one_one, antmod_picture, antmod_one_one.c.antmod_name == antmod_picture.c.antmod,isouter=True
                     ).where(antmod_one_one.c.model_name.like("%" + model_name + "%")).group_by(antmod_one_one.c.id)
        qry = conn.execute(stmt, execution_options={"prebuffer_rows": True})
        return qry


this is my modify  my code  but it still error

**Reply by q1165600785** [reply]
stmt = select(antmod_one_one.c.id, antmod_one_one.c.model_name,
                       antmod_one_one.c.antmod_name,antmod_one_one.c.rad,
                       antmod_one_one.c.dh, antmod_one_one.c.L1_E, antmod_one_one.c.L1_N,
                       antmod_one_one.c.L1_U,antmod_one_one.c.L2_E,
                       antmod_one_one.c.L2_N, antmod_one_one.c.L2_U,
                       antmod_one_one.c.remark,
                      # func.concat_ws(",",
                      func.group_concat(antmod_picture.c.picture_dir).label("antmod_dir"),
                      func.group_concat(model_picture.c.picture_dir).label("model_dir")
                      # ).label("picture_dir")
                      ).where(antmod_one_one.c.model_name.like("%" + model_name + "%"
                     )).join_from(antmod_one_one, antmod_picture, antmod_one_one.c.antmod_name == antmod_picture.c.antmod, isouter=True
                      ).join_from(antmod_one_one, model_picture, antmod_one_one.c.model_name == model_picture.c.model,isouter=True
                     ).group_by(antmod_one_one.c.id)
        qry = conn.execute(stmt, execution_options={"prebuffer_rows": True})
        return qry

this sql expression  is right  but  the field antmod_dir model_dir has two   i know  what happen result out

**Reply by q1165600785** [reply]
2023-04-01 14:28:48,428 INFO sqlalchemy.engine.Engine SELECT antmod_one_one.id, antmod_one_one.model_name, antmod_one_one.antmod_name, antmod_one_one.rad, antmod_one_one.dh, antmod_one_one.`L1_E`, antmod_one_one.`L1_N`, antmod_one_one.`L1_U`, antmod_one_one.`L2_E`, antmod_one_one.`L2_N`, antmod_one_one.`L2_U`, antmod_one_one.remark, group_concat(antmod_picture.picture_dir) AS antmod_dir, group_concat(model_picture.picture_dir) AS model_dir 
FROM antmod_one_one LEFT OUTER JOIN antmod_picture ON antmod_one_one.antmod_name = antmod_picture.antmod LEFT OUTER JOIN model_picture ON antmod_one_one.model_name = model_picture.model 
WHERE antmod_one_one.model_name LIKE %(model_name_1)s GROUP BY antmod_one_one.id
2023-04-01 14:28:48,428 INFO sqlalchemy.engine.Engine [generated in 0.00026s] {'model_name_1': '%zz%'}
(1, 'zz', '55', None, None, None, None, None, None, None, None, 'hh', '88,zz,88,zz', 'dd,dd,yy,yy')
{'id': 1, 'model_name': 'zz', 'antmod_name': '55', 'rad': None, 'dh': None, 'L1_E': None, 'L1_N': None, 'L1_U': None, 'L2_E': None, 'L2_N': None, 'L2_U': None, 'remark': 'hh', 'antmod_dir': '88,zz,88,zz', 'model_dir': 'dd,dd,yy,yy'}
2023-04-01 14:28:48,429 INFO sqlalchemy.engine.Engine COMMIT
<sqlalchemy.engine.cursor.CursorResult object at 0x00000227AA219160>

Process finished with exit code 0

**Reply by q1165600785** [reply]
i  want  to  knwo   why  restult  out  this situation happen

**Comment by CaselIT** [comment]
What's different from https://github.com/sqlalchemy/sqlalchemy/discussions/9579 ?

I replied there
